src/Sylius/Component/Promotion/Repository/PromotionCouponRepositoryInterface.php

```diff
@@ -32,4 +33,12 @@ public function createQueryBuilderByPromotionId($promotionId);
      * @return int
      */
     public function countByCodeLength($codeLength);
+
+    /**
+     * @param string $promotionCode
+     * @param string $code
+     *
+     * @return PromotionInterface
+     */
+    public function findOneByPromotionCodeAndCode($promotionCode, $code);
```

Comment:
I just a matter of convenience and narrowing a scope for me. Other functions are described this way also.